code/isi_segmentation/metadata.py

```diff
@@ -1,20 +1,112 @@
 import datetime
+import json
 from pathlib import Path
 
 from aind_data_schema.core.data_description import DataDescription, DataLevel
-from aind_data_schema.core.processing import Processing
+from aind_data_schema.core.processing import (DataProcess, PipelineProcess,
+                                              Processing)
 from aind_data_schema.core.quality_control import (QCEvaluation, QCMetric,
                                                    QCStatus, QualityControl,
                                                    Stage, Status)
 from aind_data_schema_models.modalities import Modality
+from aind_metadata_upgrader.data_description_upgrade import \
+    DataDescriptionUpgrade
 
 
-def make_data_description():
-    return None
+def make_data_description(input_dir="../data") -> DataDescriptionUpgrade:
+    """Read data_description.json from input directory.
 
+    Parameters
+    ----------
+    input_dir : str
+        Directory where data_description.json is located. Defaults to "../data".
 
-def make_processing():
-    return None
+    Returns
+    -------
+    DataDescriptionUpgrade
+        An upgraded DataDescriptionUpgrade object based on the existing data description.
+    """
+    data_description = None
+
+    if input_dir:
+        data_desc_path = Path(input_dir) / "data_description.json"
+        if data_desc_path.exists():
+            with open(data_desc_path, "r") as f:
+                data_description = json.load(f)
+    data_description_upgrader = DataDescriptionUpgrade(
```

Comment:
We'll want to pull this from the data_description coming in. Not sure if we need to handle that now until we have a real example.